tools/pi_db_tools/delta_pi_calc.xml

```diff
@@ -0,0 +1,121 @@
+<tool id="calc_delta_pi" name="Add delta pI" version="1.0">
+    <description>to peptide table</description>
+    <command>
+	    python3 $__tool_directory__/peptide_pi_annotator.py -i '$trainingpi' -p '$peptable'
+            --stripcol $stripcol --pepcol $pepcol --fraccol $fraccol --out '$output'
+	    
+	    --strippatterns
+	    #for $strip in $strips
+	        '$strip.pattern'
+	    #end for
+
+	    --intercepts 
+	    #for $strip in $strips
+	        $strip.intercept 
+	    #end for
+	    
+	    --widths
+	    #for $strip in $strips
+		$strip.fr_width
+	    #end for
+	    
+	    #if len($ignoremods) > 0
+	        --ignoremods
+	        #for $mod in $ignoremods
+		    '$mod.regex'
+		#end for
+	    #end if
+    </command>
+    
+    <inputs>
+      <param name="trainingpi" type="data" format="tabular" label="Known peptides with annotation of pI" help="First column is sequence, second pI" />
+      <param name="peptable" type="data" format="tabular" label="Peptide table to determine pI shift from" />
+      <repeat name="ignoremods" title="Peptide modification weights to ignore">
+	      <param name="regex" label="Regex to strip modification from peptide table before looking up sequence in predicted pI" type="text" help="Enter e.g. 15.994915 for oxidation, * to ignore all modifications, or a proper regex (however brackets are stripped by galaxy)" />
+      </repeat>
+      <param name="pepcol" type="integer" value="" label="Peptide sequence column in peptide table" />
+      <param name="fraccol" type="integer" value="" label="Fraction number column in peptide table" />
+      <param name="stripcol" type="integer" value="" label="Strip pattern column in peptide table" help="E.g. column with filename to derive strip name from"/>
+      <repeat name="strips" title="pI separation strip data">
+	      <param name="pattern" type="text" label="Strip regex detection pattern" />
+	      <param name="intercept" type="text" label="pI intercept of strip" />
+	     <param name="fr_width" type="text" label="fraction widths" />
+     </repeat>
+    </inputs>
+    
+    <outputs>
+        <data format="tabular" name="output"/>
+    </outputs>
+    <tests>
+	    <test>
+		    <param name="trainingpi" value="predicted_peptides.txt" />
+		    <param name="peptable" value="peptable.txt" />
+		    <repeat name="ignoremods">
+			    <param name="regex" value="*" />
```

Comment:
Think so yes. I'm not a cracker, I don't have enough evil skills to think of all evil hacks that people can do with this ;)